orchagent/vxlanorch.cpp

```diff
@@ -2347,6 +2347,14 @@ bool EvpnRemoteVnip2pOrch::addOperation(const Request& request)
         return true;
     }
 
+    EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
```

Comment:
Line 2373 has this same check and hence it can be removed from there if this change needs to go in.

Reply:
Line 2373 will not hit in this case, as it early return at line 2364.

Reply:
@jcaiMR Currently the check for the vtep ptr is present only in line 2373 under a couple of if conditions. 

Since the check is now being moved to line 2350 outside of the if checks why do we need the same check in line 2373 ?

    if (tunnel_orch->getTunnelPort(remote_vtep,tunnelPort))
    {
        SWSS_LOG_INFO("Vxlan tunnelPort exists: %s", remote_vtep.c_str());

        if (gPortsOrch->isVlanMember(vlanPort, tunnelPort))
        {

We can remove the following check as it is replicated in line 2350. 

            _EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
            auto vtep_ptr = evpn_orch->getEVPNVtep();
            if (!vtep_ptr)
            {
                SWSS_LOG_WARN("Remote VNI add: VTEP not found. remote=%s vid=%d",
                              remote_vtep.c_str(),vlan_id);
                return true;
            }_

Reply:
> @jcaiMR Currently the check for the vtep ptr is present only in line 2373 under a couple of if conditions.
> 
> Since the check is now being moved to line 2350 outside of the if checks why do we need the same check in line 2373 ?
> 
> ```
> if (tunnel_orch->getTunnelPort(remote_vtep,tunnelPort))
> {
>     SWSS_LOG_INFO("Vxlan tunnelPort exists: %s", remote_vtep.c_str());
> 
>     if (gPortsOrch->isVlanMember(vlanPort, tunnelPort))
>     {
> ```
> 
> We can remove the following check as it is replicated in line 2350.
> 
> ```
>         _EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
>         auto vtep_ptr = evpn_orch->getEVPNVtep();
>         if (!vtep_ptr)
>         {
>             SWSS_LOG_WARN("Remote VNI add: VTEP not found. remote=%s vid=%d",
>                           remote_vtep.c_str(),vlan_id);
>             return true;
>         }_
> ```

I see, will update in latest diff. And Rajesh, I plan to use this PR as a workaround to make warmboot function pass first. Will raise another PR for the final root cause fix.

Reply:
Diff updated.